core/cmd/chains_commands.go

```diff
@@ -0,0 +1,225 @@
+	}
+}
+
+// IndexChains returns all chains.
+func (cli *chainClient[C, R, P, P2]) IndexChains(c *cli.Context) (err error) {
+	var p P2
+	return cli.getPage(cli.path, c.Int("page"), &p)
+}
+
+// CreateChain adds a new  chain.
+func (cli *chainClient[C, R, P, P2]) CreateChain(c *cli.Context) (err error) {
+	chainID := c.String("id")
+	if chainID == "" {
+		return cli.errorOut(errors.New("missing chain ID [-id]"))
+	}
+	if !c.Args().Present() {
+		return cli.errorOut(errors.New("must pass in the chain config parameters [JSON blob | JSON filepath]"))
+	}
+
+	buf, err := getBufferFromJSON(c.Args().First())
+	if err != nil {
+		return cli.errorOut(err)
+	}
+
+	params := map[string]interface{}{
+		"chainID": chainID,
+		"config":  json.RawMessage(buf.Bytes()),
+	}
+
+	body, err := json.Marshal(params)
+	if err != nil {
+		return cli.errorOut(err)
+	}
+
+	var resp *http.Response
+	resp, err = cli.HTTP.Post(cli.path, bytes.NewBuffer(body))
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	defer func() {
+		if cerr := resp.Body.Close(); cerr != nil {
+			err = multierr.Append(err, cerr)
+		}
+	}()
+
+	var p P
+	return cli.renderAPIResponse(resp, &p)
+}
+
+// RemoveChain removes a specific  Chain by id.
+func (cli *chainClient[C, R, P, P2]) RemoveChain(c *cli.Context) (err error) {
+	if !c.Args().Present() {
+		return cli.errorOut(errors.New("must pass the id of the chain to be removed"))
+	}
+	chainID := c.Args().First()
+	resp, err := cli.HTTP.Delete(cli.path + "/" + chainID)
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	_, err = cli.parseResponse(resp)
+	if err != nil {
+		return cli.errorOut(err)
+	}
+
+	fmt.Printf("Chain %v deleted\n", chainID)
+	return nil
+}
+
+// ConfigureChain configures an existing  chain.
+func (cli *chainClient[C, R, P, P2]) ConfigureChain(c *cli.Context) (err error) {
+	chainID := c.String("id")
+	if chainID == "" {
+		return cli.errorOut(errors.New("missing chain ID [-id]"))
+	}
+
+	if !c.Args().Present() {
+		return cli.errorOut(errors.New("must pass in at least one chain configuration parameters (usage: ... chains configure [key1=value1 key2=value2 ...])"))
+	}
+
+	// Fetch existing config
+	resp, err := cli.HTTP.Get(fmt.Sprintf("%s/%v", cli.path, chainID))
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	defer func() {
+		if cerr := resp.Body.Close(); cerr != nil {
+			err = multierr.Append(err, cerr)
+		}
+	}()
+	var chain R
+	if err = cli.deserializeAPIResponse(resp, &chain, &jsonapi.Links{}); err != nil {
+		return cli.errorOut(err)
+	}
+	config := chain.GetConfig()
+
+	// Parse new key-value pairs
+	params := map[string]interface{}{}
+	for _, arg := range c.Args() {
+		parts := strings.SplitN(arg, "=", 2)
+		if len(parts) != 2 {
+			return cli.errorOut(errors.Errorf("invalid parameter: %v", arg))
+		}
+
+		var value interface{}
+		if err = json.Unmarshal([]byte(parts[1]), &value); err != nil {
+			// treat it as a string
+			value = parts[1]
+		}
+		// TODO: handle `key=nil` and `key=` besides just null?
```

Comment:
Shall we allow `nil`? I would rather `cli.errorOut` instead

Reply:
:shrug: I actually have no context for this - just moved the pre-existing code.